Source/inv.cpp

```diff
@@ -817,17 +817,19 @@ void CheckInvPaste(int pnum, Point cursorPosition)
 		if (i >= InvRect[r].x + xo && i <= InvRect[r].x + xo + INV_SLOT_SIZE_PX) {
 			if (j >= InvRect[r].y + yo - INV_SLOT_SIZE_PX - 1 && j < InvRect[r].y + yo) {
 				done = true;
-				r--;
 			}
 		}
-		if (r == SLOTXY_CHEST_LAST) {
+		if (r == SLOTXY_INV_FIRST) {
 			if ((sx & 1) == 0)
-				i -= 14;
+				i -= INV_SLOT_HALF_SIZE_PX;
 			if ((sy & 1) == 0)
-				j -= 14;
+				j -= INV_SLOT_HALF_SIZE_PX;
 		}
-		if (r == SLOTXY_INV_LAST && (sy & 1) == 0)
-			j += 14;
+		if (r == SLOTXY_BELT_FIRST && (sy & 1) == 0)
+			j += INV_SLOT_HALF_SIZE_PX;
+
+		if (done) // found a slot
+			break;
 	}

```

Comment:
`for`-loop already checks for `done`.

```suggestion
```

Reply:
yes but I don't want it there, I guess I forgot to remove